hoodie-client/src/main/java/com/uber/hoodie/HoodieReadClient.java

```diff
@@ -57,24 +60,21 @@
    * just with a simple basepath pointing to the dataset. Until, then just always assume a
    * BloomIndex
    */
-  private transient final HoodieBloomIndex index;
+  private transient final HoodieIndex<T> index;
   private final HoodieTimeline commitTimeline;
-  private HoodieTable hoodieTable;
+  private transient HoodieTable hoodieTable;
   private transient Optional<SQLContext> sqlContextOpt;
 
   /**
    * @param basePath path to Hoodie dataset
    */
   public HoodieReadClient(JavaSparkContext jsc, String basePath) {
-    this.jsc = jsc;
-    this.fs = FSUtils.getFs(basePath, jsc.hadoopConfiguration());
-    // Create a Hoodie table which encapsulated the commits and files visible
-    this.hoodieTable = HoodieTable
-        .getHoodieTable(new HoodieTableMetaClient(jsc.hadoopConfiguration(), basePath, true), null);
-    this.commitTimeline = hoodieTable.getCommitTimeline().filterCompletedInstants();
-    this.index =
-        new HoodieBloomIndex(HoodieWriteConfig.newBuilder().withPath(basePath).build(), jsc);
-    this.sqlContextOpt = Optional.absent();
+    this(jsc, HoodieWriteConfig.newBuilder()
+        .withPath(basePath)
+        // by default we use HoodieBloomIndex
```

Comment:
would n't this be different for the actual use case? HBaseIndex?

Reply:
This is refactoring of an existing Constructor to use a new Constructor that takes HoodieWriteConfig.  Going forward, client should use the constructor so that the proper HoodieIndex will be created.

Reply:
sg